### PR TITLE
Transform JSON to conform to persistence schema

### PR DESCRIPTION
**Pull request by vrajmohan**

### CODE REVIEW

File: src/utils/jsonTransform.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import {isAffirmative} from './utils';
+
+export function to_registration_service_format(src) {
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: lets use camelCase for JS methods

Comment (el-mapache) [inline src/utils/jsonTransform.js]: Here I think we can define the function, and then call `export default ${fnname}`. We can always add additional exports as necessary.

File: src/utils/jsonTransform.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    };
+}
+
+function assemble_household(src) {
+    return src.household.members.map( member => {
+        return {
+            first_name: member.name.firstName,
+            middle_name: member.name.middleName,
+            last_name: member.name.lastName,
+            dob: `${member.dob.year}-${member.dob.month}-${member.dob.day}`, // If we are going to validate it, can we make it available as a single field TODO
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: We can add a method in the data model that formats the data in the way the server expects, although having the transform here seems fine.

File: src/utils/jsonTransform.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            sex: member.sex,
+            ssn: member.ssn,
+            race: member.race,
+            has_food_assistance: member.hasFoodAssistance, // What is this used for? TODO
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: Not sure, @edmullen flagged it as necessary. We should make an issue to raise it.

File: src/utils/jsonTransform.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            money_on_hand: member.assetsAndIncome.moneyOnHand, // Why is this a string? TODO
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: I don't do any type conversions, and form data is always provided by the browser as a string. We can always call `Number(value)` to force a conversion.

File: src/utils/jsonTransform.test.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
+import {to_registration_service_format} from './jsonTransform';
```
Comment (el-mapache) [inline src/utils/jsonTransform.test.js]: file naming conventions in this project use hyphens. We should change the name to `json-transform`.

File: src/utils/jsonTransform.test.js [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+const testData = {
```
Comment (el-mapache) [inline src/utils/jsonTransform.test.js]: Here we can import the test data file, just like we import the formatting method.

File: src/utils/jsonTransform.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            jobs: assemble_jobs(member),
+        }
+    });
+}
+
+function assemble_jobs(member) {
+    return member.assetsAndIncome.jobs.map(job => {
+        return {
+            employer_name: job.employerName,
+            pay: job.pay, // TODO this is a string?
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: It is a string, see comment on line 38

File: src/utils/jsonTransform.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                 alimony: member.assetsAndIncome.incomeSources.alimony.value,
+                 child_support: member.assetsAndIncome.incomeSources.childSupport.value,
+                 other_sources: member.assetsAndIncome.incomeSources.otherSources.value,
+            },
+            jobs: assemble_jobs(member),
+        }
+    });
+}
+
+function assemble_jobs(member) {
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: see comment on line 3

File: src/utils/jsonTransform.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            home_or_business_repairs: src.impact.otherExpenses.repairs.value,
+            temporary_shelter_expenses: src.impact.otherExpenses.tempShelter.value,
+            evacuation_expenses: src.impact.otherExpenses.evacuation.value,
+            other: src.impact.otherExpenses.other.value
+        },
+        household: assemble_household(src)
+    };
+}
+
+function assemble_household(src) {
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: see comment on line 3

File: src/utils/jsonTransform.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+export function to_registration_service_format(src) {
+    return {
+        disaster_id: src.basicInfo.disasterIndex, // exists in UI debug, but not in test-data.json TODO
+        // "preferred_language": TODO
+        phone: src.basicInfo.phone,
+        email: src.basicInfo.email,
+        residential_address: src.basicInfo.residenceAddress,
+        mailing_address: src.basicInfo.mailingAddress,
+        county: src.basicInfo.county, // should it be disasterCounty?
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: I'm not entirely sure, I thought the guidance was that the user could enter whatever county they resided in, 
regardless of whether or not that county was affected by the disaster. If that isn't the case, we don't need to ask the user for this data, we can use the value they've selected during the `pre-registration` step.

We should make an issue to have this clarified and documented.

File: src/utils/jsonTransform.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
+import {isAffirmative} from './utils';
+
+export function to_registration_service_format(src) {
+    return {
+        disaster_id: src.basicInfo.disasterIndex, // exists in UI debug, but not in test-data.json TODO
```
Comment (el-mapache) [inline src/utils/jsonTransform.js]: This field has been added to the `test-data.js` file

**Comment by vrajmohan** [comment]
All review comments have been addressed.

**Pull request merged**